### PR TITLE
Update to 5.2.1

### PR DESCRIPTION
**Pull request by matthiasdiener**
Continuation of #33.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthiasdiener** [comment]
@conda-forge-admin, please rerender

**Comment by matthiasdiener** [comment]
@conda-forge-admin restart ci

**Comment by minrk** [comment]
@conda-forge-admin please rerender

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
@conda-forge-admin, please restart ci

**Comment by minrk** [comment]
There are [a number of upstream PRs](https://github.com/KarypisLab/METIS/pulls) attempting to modernize the CMake configurations to various degrees. I wonder if it makes sense to rely on one or more of those.

**Comment by minrk** [comment]
A few more patches and all builds are now passing. Anyone with more expertise with metis know if there's something more to test?

**Comment by mikemhenry** [comment]
@zklaus can you take a look at this?

File: recipe/getprocvmpeak.diff
Comment (zklaus) [inline recipe/getprocvmpeak.diff:1]: I suggest to align this with the other patches by moving to `patches/getprocvmpeak.patch`.

**Review by zklaus — APPROVED** [review]
I have only one small stylistic comments, otherwise it looks good to me.
@conda-forge/metis, could you please have a final look and merge if appropriate?

**Comment by jaimergp** [comment]
Ooops:

```
+ mpmetis graphs/metis.mesh 10
/home/conda/feedstock_root/build_artifacts/metis_1708951629302/test_tmp/run_test.sh: line 5:  1223 Illegal instruction     (core dumped) mpmetis graphs/metis.mesh 10
```

**Comment by jaimergp** [comment]
Might be unrelated and just a non-deterministic segfault.

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
The illegal instruction seems to be in the GKLibs build. Stacktrace from local debug build:
```
(gdb) run
Starting program: /home/conda/feedstock_root/build_artifacts/debug_1708961928483/work/build/programs/mpmetis 
warning: Error disabling address space randomization: Operation not permitted
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Program received signal SIGILL, Illegal instruction.
0x00007fb54036cfca in gk_malloc () from /home/conda/feedstock_root/build_artifacts/debug_1708961928483/_h_env/lib/libGKlib.so
(gdb) where
#0  0x00007fb54036cfca in gk_malloc () from /home/conda/feedstock_root/build_artifacts/debug_1708961928483/_h_env/lib/libGKlib.so
#1  0x00005637f720d9e4 in parse_cmdline (argc=1, argv=0x7ffe1437f748) at /usr/local/src/conda/metis-5.2.1/programs/cmdline_mpmetis.c:200
#2  0x00005637f720d5b5 in main (argc=<optimized out>, argv=<optimized out>) at /usr/local/src/conda/metis-5.2.1/programs/mpmetis.c:31
```

**Comment by jaimergp** [comment]
IIRC, @hmacdope observed this already, and was able to build it successfully with the latest `GKlib` (https://github.com/KarypisLab/GKlib/commit/8bd6bad750b2b0d90800c632cf18e8ee93ad72d7). What if we build GKlib from source in this repo (there're no recent tagged releases) and we get back to linking it statically?

**Comment by zklaus** [comment]
Yeah, that's my conclusion too. I did build the older 5.1.1 version GKLib and used it successfully in the new 5.2.1 build of METIS. I am pretty sure the problem that we are seeing here is due to the use of `-march=native` in the 5.1.1 build of gklib on conda-forge, but seeing as it is quite old anyway, I'll set up a build here with static gklib at current master (plus patch to get rid of `-march=native`).

**Comment by zklaus** [comment]
The static approach is in #43.

**Pull request closed without merging**